src/transactions/transactionBytes.js

```diff
@@ -203,132 +148,110 @@ function getEmptyBytesForTransaction(transaction) {
-	 * @param transactionBuffer buffer
-	 * @param assetSize number
-	 * @param assetBytes number
-	 * @return {buffer}
-	 */
+		this.transactionSenderPublicKey = Buffer.from(transaction.senderPublicKey, 'hex');
+		this.transactionRequesterPublicKey = transaction.requesterPublicKey
+			? Buffer.from(transaction.requesterPublicKey, 'hex')
+			: Buffer.alloc(0);
 
-	function assignTransactionBuffer(transactionBuffer, assetSize, assetBytes) {
-		transactionBuffer.writeInt8(transaction.type);
-		transactionBuffer.writeInt(transaction.timestamp);
+		this.transactionRecipientID = transaction.recipientId
+			? Buffer.from(
+				bignum(
+					transaction.recipientId.slice(0, -1),
+				).toBuffer({ size: this.byteSizes.RECIPIENT_ID }),
+			)
+			: Buffer.alloc(this.byteSizes.RECIPIENT_ID).fill(0);
 
-		assignHexToTransactionBytes(transactionBuffer, transaction.senderPublicKey);
+		this.transactionAmount = Buffer.alloc(this.byteSizes.AMOUNT);
+		this.transactionAmount.writeInt32LE(transaction.amount, 0, this.byteSizes.AMOUNT);
 
-		if (transaction.requesterPublicKey) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.requesterPublicKey);
-		}
+		this.transactionAssetData = this.getAssetBytes(transaction);
 
-		if (transaction.recipientId) {
-			let recipient = transaction.recipientId.slice(0, -1);
-			recipient = bignum(recipient).toBuffer({ size: 8 });
+		this.transactionSignature = transaction.signature
+			? Buffer.from(transaction.signature, 'hex')
+			: Buffer.alloc(0);
 
-			for (let i = 0; i < 8; i++) {
-				transactionBuffer.writeByte(recipient[i] || 0);
-			}
-		} else {
-			for (let i = 0; i < 8; i++) {
-				transactionBuffer.writeByte(0);
-			}
-		}
-		transactionBuffer.writeLong(transaction.amount);
+		this.transactionSecondSignature = transaction.signSignature
+			? Buffer.from(transaction.signSignature, 'hex')
+			: Buffer.alloc(0);
 
-		if (transaction.asset.data) {
-			const dataBuffer = Buffer.from(transaction.asset.data);
-			for (let i = 0; i < dataBuffer.length; i++) {
-				transactionBuffer.writeByte(dataBuffer[i]);
-			}
-		}
-
-		if (assetSize > 0) {
-			for (let i = 0; i < assetSize; i++) {
-				transactionBuffer.writeByte(assetBytes[i]);
-			}
-		}
 
-		if (transaction.signature) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.signature);
-		}
+		this.checkTransaction();
+		return this;
+	}
 
-		if (transaction.signSignature) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.signSignature);
-		}
+	/**
+	 * @method get transactionBytes
+	 * @return {Buffer}
+	 */
 
-		transactionBuffer.flip();
-		const arrayBuffer = new Uint8Array(transactionBuffer.toArrayBuffer());
-		const buffer = [];
+	get transactionBytes() {
+		return this.concatTransactionBytes();
+	}
 
-		for (let i = 0; i < arrayBuffer.length; i++) {
-			buffer[i] = arrayBuffer[i];
-		}
+	/**
+	 * @method concatTransactionBytes
+	 * @private
+	 * @return {Buffer}
+	 */
 
-		return Buffer.from(buffer);
+	concatTransactionBytes() {
```

Comment:
Does this need to be a separate method or can we put this logic directly in `get transactionBytes`?